src/libpriv/rpmostree-postprocess.c

```diff
@@ -1506,10 +1544,56 @@ rpmostree_treefile_postprocessing (GFile         *yumroot,
         }
     }
 
+  {
+    const char *base_version = NULL;
+
+    if (!_rpmostree_jsonutil_object_get_optional_string_member (treefile,
+                                                                "mutate-os-release",
+                                                                &base_version,
+                                                                error))
+      goto out;
+
+    if (base_version != NULL)
+      {
+        glnx_fd_close int root_dfd = -1;
+        g_autofree char *contents = NULL;
+        g_autofree char *new_contents = NULL;
+        g_autofree char *path = NULL;
+
+        /* we need to follow all potential symlinks so that we replace the
+         * actual file, not directly create a file at /etc/os-release */
+        {
+          g_autofree char *tmp
+            = g_build_filename (gs_file_get_path_cached (yumroot),
+                                "etc/os-release", NULL);
+          path = realpath (tmp, NULL);
```

Comment:
In general this is dangerous, since if the symlink is absolute we'll happily walk out of the container root.  My instinct here is to search for `/usr/lib/os-release` and `/etc/os-release` and take the first non-symlink file.